lib/streaming-pipeline.js

```diff
@@ -106,22 +97,35 @@ module.exports = class StreamingPipeline extends Writable {
         }
 
         if (inputSignal.hasStart()) {
-            const outputContentTypes = inputSignal.getStart().getExpectedcontenttypesList();
+            const startSignal = inputSignal.getStart();
+            const outputContentTypes = startSignal.getExpectedcontenttypesList();
+            const inputStreamNames = startSignal.getInputnamesList();
+            const outputStreamNames = startSignal.getOutputnamesList();
             if (this._startReceived) {
                 callback(new RiffError(
                     'error-streaming-too-many-starts',
-                    `start signal has already been received. Rejecting new start signal with content types [${outputContentTypes.join()}]`));
+                    'start signal has already been received. ' +
+                    `Rejecting start signal with: output content types: [${outputContentTypes.join(",")}], ` +
+                    `input names: [${inputStreamNames.join(",")}], ` +
+                    `output names: [${outputStreamNames.join(",")}]`));
+
                 return;
             }
-            const outputCount = outputContentTypes.length;
-            if (outputCount > this.parameterCount) {
+            this._streamMetadata.inputs = inputStreamNames;
+            this._streamMetadata.outputs = outputStreamNames;
+            const actualOutputCount = outputContentTypes.length;
+            const expectedOutputCount = this._streamMetadata.outputs.length;
+            if (actualOutputCount !== expectedOutputCount) {
```

Comment:
neither of these is an "expected" value since they both come from the start signal. I'd expect the expected value to come from the function, but since that's no longer variadic, what is this even testing?

Reply:
Will change the naming. Problems are very unlikely to happen now but if they do, isn't it better to fail fast?

Reply:
Seems like a flaw in the streaming protocol design that such a situation is possible.

Reply:
Output names and expect content types could indeed be merged: I opened https://github.com/projectriff/invoker-specification/issues/8. 
I suggest we keep the validation like this until the other issue is fixed.